niceml/data/datasets/dfdataset.py

```diff
@@ -100,19 +220,34 @@ def get_all_data(self):
     def extract_data(self, cur_indexes, cur_input):
         """extracts data"""
         cur_key = cur_input["key"]
-        cur_data = self.dataframe.loc[cur_indexes, cur_key]
-        if cur_input["type"] == "categorical":
+        cur_data = self.data.iloc[cur_indexes][cur_key]
+        if cur_input["type"] == FeatureType.CATEGORICAL:
             cur_data = to_categorical(cur_data, cur_input["value_count"])
         return cur_data
 
     def __getitem__(self, index):
+        """
+        The __getitem__ function is called when the `DfDataset` is indexed (while training a model).
+
+        Args:
+            index: Specify the start index of the batch
+
+        Returns:
+            A batch of input data and target data with the batch size `self.batch_size`
+        """
         start_idx = index * self.batch_size
         end_idx = min(len(self.index_list), (index + 1) * self.batch_size)
         input_data, target_data = self.get_data(start_idx, end_idx)
 
         return input_data, target_data
 
     def __len__(self):
+        """
+        The __len__ function is used to determine the number of batches in an epoch.
```

Comment:
Does the function return the number of batches in an epoch or in the dataset? Missleading docstring.

Reply:
The info is the same. I have unified it (epoch)